set.mm

```diff
@@ -424459,6 +424480,51 @@ then they have the same degree (i.e., the same number of adjacent
       BVSURUSUTVAVBUNSWJWAVFWJVQWAVQWIULTWSRVCVDWKDVGJZWGPZWOWMOWJXFWAVQXEWIWGD
       VOZWTVHTVSDEFVIVJWKXEXAPZWQWNOWJXHWAVQXEWIXAXGXBVHTVRDEFVIVJVKVSCWLGVLVRC
       WLGVLVMVPVN $.
+
+    $d D x y $.  $d X x y $.  $d Y x y $.
+    frgrwopreglem4a.e $e |- E = ( Edg ` G ) $.
+    $( In a friendship graph each vertex is connected with a vertex with a
+       different degree.  Alternate version of ~ frgrwopreglem4 without a fixed
+       degree and without using the sets ` A ` and ` B ` .  (Contributed by
+       Alexander van der Vekens, 30-Dec-2017.)  (Revised by AV, 4-Feb-2022.) $)
+    frgrwopreglem4a $p |- ( ( G e. FriendGraph /\ ( X e. V /\ Y e. V )
+                                               /\ ( D ` X ) =/= ( D ` Y ) )
+                            -> { X , Y } e. E ) $=
+      ( vy vx wcel wa cfv wne wceq wi fveq2 cnbgr co w3a cpr a1i necon3d imp cv
+      cfrgr 3adant1 wnel csn cdif wral frgrncvvdeq wb neleq2 syl eqeq1d imbi12d
+      oveq2 neleq1 eqeq2d simpll sneq difeq2d adantl simpr necom biimpi anim12i
+      eldifsn sylibr rspc2vd wn nnel nbgrsym frgrusgr nbusgreledg biimpd sylbid
+      cusgr a1d expcom sylbi eqneqall 2a1d ja com12 syld com3l mpcom expd com34
+      3imp mpd ) CUGLZEDLZFDLZMZEANZFANZOZUAEFOZEFUBBLZWRXAXBWOWRXAXBWREFWSWTEF
+      PWSWTPZQWREFARUCUDUEUHWOWRXAXBXCQWOWRXBXAXCWOWRXBXAXCQZJUFZCKUFZSTZUIZXGA
+      NZXFANZPZQZJDXGUJZUKZULKDULZWOWRXBMZXEQKJACDGHUMXQXPWOXEXQXPFCESTZUIZXDQZ
+      WOXEQZXQXTXFXRUIZWSXKPZQXMKJEFDXODEUJZUKZXGEPZXIYBXLYCYFXHXRPXIYBUNXGECSU
+      SXHXRXFUOUPYFXJWSXKXGEARUQURXFFPZYBXSYCXDXFFXRUTYGXKWTWSXFFARVAURWPWQXBVB
+      YFXOYEPXQYFXNYDDXGEVCVDVEXQWQFEOZMFYELWRWQXBYHWPWQVFXBYHEFVGVHVIFDEVJVKVL
+      XTXQYAXSXDXQYAQZXSVMFXRLZYIFXRVNYJYAXQWOYJXEWOYJMXCXAWOYJXCWOYJECFSTLZXCC
+      EFUGVOWOYKXCWOCVTLYKXCUNCVPBCFEIVQUPVRVSUEWAWBWAWCXDXEXQWOXCWSWTWDWEWFWGW
+      HWIWJWKWLWMWN $.
+
+    $( If a friendship graph has two vertices with the same degree and two
+       vertices with different degrees, then there is a 4-cycle in the graph.
```

Comment:
"... with the same degree and two other vertices with degrees different from that common degree, then these four vertices form a 4-cycle."

Reply:
I will change that comment slightly...